src/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,223 @@
+  const [titleEdit, setTitleEdit] = useState(activeTodo.title);
+  const [activeTitle, setActiveTitle] = useState(activeTodo.title);
+  const [updating, setUpdating] = useState(false);
+
+  const inputFocus = useRef<HTMLInputElement>(null);
+
+  const updatingTodoOnServer = () => {
+    todoService.updateTodo(activeTodo)
+      .then((updatedTodo) => {
+        const newTodos = [...todos];
+        const index = newTodos.findIndex(t => t.id === updatedTodo.id);
+
+        newTodos.splice(index, 1, updatedTodo);
+        setTodos(newTodos);
+      })
+      .catch(() => {
+        errorHandler(ErrorMessage.update);
+        setTitleEdit(activeTodo.title);
+        activeTodo.title = activeTitle;
+        setActiveTitle(activeTitle);
+        setTitleEdit(activeTitle);
+      })
+      .finally(() => {
+        setUpdating(false);
+        setSelectedTodo(0);
+      });
+    setEditing(false);
+  };
+
+  const activeLoader = (selectedTodo === activeTodo.id && isDeliting)
+    || (selectedTodo === activeTodo.id && isSubmitting)
+    || (clear && todo.completed)
+    || (updating && selectedTodo === activeTodo.id);
+
+  const handleCheckbox = () => {
+    setUpdating(true);
+    setSelectedTodo(activeTodo.id);
+    if (!activeTodo.completed) {
+      activeTodo.completed = true;
+    } else {
+      activeTodo.completed = false;
+    }
+
+    todoService.updateTodo(activeTodo)
+      .then((updatedTodo) => {
+        const newTodos = [...todos];
+        const index = newTodos.findIndex(t => t.id === updatedTodo.id);
+
+        newTodos.splice(index, 1, updatedTodo);
+        setTodos(newTodos);
+      })
+      .catch(() => {
+        errorHandler(ErrorMessage.update);
+        if (!activeTodo.completed) {
+          activeTodo.completed = true;
+        } else {
+          activeTodo.completed = false;
+        }
+      })
+      .finally(() => {
+        setUpdating(false);
+        setSelectedTodo(0);
+      });
+  };
+
+  const isChecked = activeTodo.completed;
+
+  useEffect(() => {
+    if (editing && inputFocus.current) {
+      inputFocus.current.focus();
+    }
+  }, [editing]);
+
+  const handleDoubleClick = () => {
+    setTitleEdit(titleEdit.trim());
+    setEditing(true);
+  };
+
+  const handleEdit = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleEdit(event.target.value);
+  };
+
+  const handleKey = (event: React.KeyboardEvent) => {
+    event.preventDefault();
+
+    if (event.key === 'Enter') {
+      setSelectedTodo(activeTodo.id);
+      setUpdating(true);
+
+      if (!titleEdit.length) {
+        return deleteTodo(activeTodo.id);
+      }
+
+      activeTodo.title = titleEdit;
+
+      updatingTodoOnServer();
+    }
+
+    if (event.key === 'Escape') {
+      setEditing(false);
+      setTitleEdit(activeTodo.title);
+    }
+
+    return setTodos(todos);
+  };
+
+  const handleBlur = () => {
+    setSelectedTodo(activeTodo.id);
+    setUpdating(true);
+    if (!titleEdit.length) {
```

Comment:
```suggestion
    if (!titleEdit.trim().length) {
```
better to delete todo in this case

https://github.com/mate-academy/react_todo-app-with-api/assets/70220667/20a6e2c1-2717-4ade-be5b-b73f604b9ba9